process_request/helpers.py

```diff
@@ -3,6 +3,70 @@
 from rapidfuzz import fuzz
 
 
+def get_collection_creator(archival_object):
+    """Takes json for an archival object that has the _resolved parameter on resource::linked_agents. Iterates through linked_agents; if the role is creator, appends to list, and returns list as a string."""
+    creators = []
+    for linked_agent in archival_object.get("resource").get("_resolved").get("linked_agents"):
+        if linked_agent.get("role") == "creator":
+            creators.append(linked_agent.get("_resolved").get('display_name').get('sort_name'))
+    return ",".join(creators)
+
+
+def get_location(top_container_info):
+    """Takes the _resolved json for a top container (with the _resolved parameter also on container locations) from an archival object and returns the title field of the container location for each location. C"""
+    locations = []
+    for c in top_container_info.get("container_locations"):
+        locations.append(c.get("_resolved").get("title"))
+    return ",".join(locations)
+
+
+def get_container_info(top_container_info, container_field):
+    """Takes two arguments: the _resolved json for a top container from an archival objects, and the top container field to retrieve."""
+    return(top_container_info.get(container_field))
+
+
+def check_for_instance_type(archival_object, type_to_check):
+    """Takes json for an archival object and instance type to check against. If the instance type is found, return the index of the instance"""
+    list_of_instances = []
+    for i in archival_object.get("instances"):
+        instance_type = i.get("instance_type")
+        list_of_instances.append(instance_type)
+    if type_to_check in list_of_instances:
+        return list_of_instances.index(type_to_check)
```

Comment:
this falls under @p-galligan's area of work but I think this function should just return the relevant instances, rather than an index which you then have to use in another method to get the instance.